corehq/apps/userreports/tasks.py

```diff
@@ -120,51 +119,46 @@ def resume_building_indicators(indicator_config_id, initiated_by=None):
     with notify_someone(initiated_by, success_message=success, error_message=failure, send=send):
         resume_helper = DataSourceResumeHelper(config)
 
-        relevant_ids = resume_helper.get_ids_to_resume_from()
-        if len(relevant_ids) > 0:
-            _build_indicators(config, get_document_store(config.domain, config.referenced_doc_type), relevant_ids,
-                              resume_helper)
-            last_id = relevant_ids[-1]
-            _iteratively_build_table(config, last_id, resume_helper)
+        _iteratively_build_table(config, resume_helper)
 
 
-@task(queue=UCR_CELERY_QUEUE, ignore_result=True)
-def recalculate_indicators(indicator_config_id, initiated_by=None):
-    config = _get_config_by_id(indicator_config_id)
-    adapter = get_indicator_adapter(config)
-    doc_id_provider = adapter.get_distinct_values('doc_id', 10000)[0]
-    rebuild_indicators_in_place(indicator_config_id, initiated_by, doc_id_provider)
-
-
-def _iteratively_build_table(config, last_id=None, resume_helper=None,
-                             in_place=False, doc_id_provider=None, limit=-1):
+def _iteratively_build_table(config, resume_helper=None, in_place=False,
+                             doc_id_provider=None, limit=-1):
     resume_helper = resume_helper or DataSourceResumeHelper(config)
     indicator_config_id = config._id
-    case_type_or_xmlns = config.get_case_type_or_xmlns_filter()
+    case_type_or_xmlns_list = config.get_case_type_or_xmlns_filter()
+    completed_ct_xmlns = resume_helper.get_completed_case_type_or_xmlns()
+    if completed_ct_xmlns:
+        case_type_or_xmlns_list = [
+            case_type_or_xmlns
+            for case_type_or_xmlns in case_type_or_xmlns_list
+            if case_type_or_xmlns not in completed_ct_xmlns
+        ]
 
-    relevant_ids = []
-    document_store = get_document_store(
-        config.domain, config.referenced_doc_type, case_type_or_xmlns=case_type_or_xmlns
-    )
+    for case_type_or_xmlns in case_type_or_xmlns_list:
+        relevant_ids = []
+        document_store = get_document_store(
+            config.domain, config.referenced_doc_type, case_type_or_xmlns=case_type_or_xmlns
+        )
 
-    if not doc_id_provider:
-        doc_id_provider = document_store.iter_document_ids(last_id)
+        if not doc_id_provider:
+            doc_id_provider = document_store.iter_document_ids()
```

Comment:
I think this will only ever happen on the first loop since after that it will always have a value. That seems like it will be an issue since each loop will have a different `case_type` / `xmlns` filter.

Reply:
It was only used in one place (custom mgmt command) and it's no longer needed

https://github.com/dimagi/commcare-hq/pull/18460/commits/5f1769cce65cccef77e8d70bb4ce1ec9d8ace27c